File: Lesson_6/HomeTask/Task_1/CalculatorPresenter.java
package Lesson_6.HomeTask.Task_1;

public class CalculatorPresenter {
    private CalculatorView view;
    private CalculatorModel model;

    public CalculatorPresenter(CalculatorView view, CalculatorModel model) {
        this.view = view;
        this.model = model;
    }

    public CalculatorPresenter() {
        return;
    }

    public void performCalculation() {
        double number = view.getDouble();
        String operator = view.getInputOperator();
        view.scan.nextLine();//строка, без которой, почему-то второй getDouble сразу закрывает скан, попадает в кетч и только после этогьо нормально захватывает
        double number_2 = view.getDouble_2();
        model.setResult(number);

        switch (operator) {
            case "+":
                model.add(number_2);
                break;
            case "-":
                model.minus(number_2);
                break;
            case "*":
                model.mult(number_2);
                break;
            case "/":
                model.division(number_2);
                break;


        }

        view.displayResult(model.getResult());

    }
}

class ExpandedCalculatorPresenter extends CalculatorPresenter {
    /*мы, без изменения основного класса расширили его функционал
(Принцип открытости/закрытости (Open Closed Principle))

так же этот дочерний класс расширяет базовый , я использовал так же
Принцип подстановки Барбары Лисков (Liskov’s Substitution Principle)
*/
    private ExpandedCalculatorView view_1;
    private ExpandedCalculatorModel model_1;

    public ExpandedCalculatorPresenter(ExpandedCalculatorView view, ExpandedCalculatorModel model) {
        this.view_1 = view;
        this.model_1 = model;
    }

    @Override
    public void performCalculation() {
        double number = view_1.getDouble();
        String operator = view_1.getInputOperator_1();
        view_1.scan.nextLine();//строка, без которой, почему-то второй getDouble сразу закрывает скан, попадает в кетч и только после этогьо нормально захватывает

        model_1.setResult(number);
        double number_2 = 0;
        switch (operator) {
            case "+":
                number_2 = view_1.getDouble_2();
                model_1.add(number_2);
                break;
            case "-":
                number_2 = view_1.getDouble_2();
                model_1.minus(number_2);
                break;
            case "*":
                number_2 = view_1.getDouble_2();
                model_1.mult(number_2);
                break;
            case "/":
                number_2 = view_1.getDouble_2();
                model_1.division(number_2);
                break;
            case "sqrt":

                model_1.sqrt();
                break;
            case "pow":
                number_2 = view_1.getInt();
                model_1.power(number_2);


        }

        view_1.displayResult(model_1.getResult());

    }
}

File: Lesson_7/Home_task/Task_2/CalculatorPresenter.py
import logging

class CalculatorPresenter:
    def __init__(self, model_1, view_1,number=90):
        self.model = model_1
        self.view = view_1
        self.operator=number
    def performCalculation(self):
        logging.basicConfig(level=logging.INFO, filename="calc_log_1.log", filemode="w")

        logger = logging.getLogger("log_1")
        number = self.view.getDouble()
        self.operator = self.view.getInputOperator()
        number_2 = self.view.getDouble_2()
        self.model.setResult(number)

        if self.operator == "+":
            self.model.add(number_2)

        elif self.operator == '-':
            self.model.minus(number_2)
        elif self.operator == "*":
            self.model.mult(number_2)
        elif self.operator == "/":
            self.model.division(number_2)
        elif self.operator=="stop":
            return
        logger.info('{0} {1} {2} = {3}'.format(number, self.operator, number_2,self.model.getResult()))
        self.view.displayResult(self.model.getResult())